### DISCUSSION TITLE
How to prevent scraping?

### DISCUSSION THREAD

**Discussion by lilouartz** [remix-run/remix#9267] [General]
I realize that there are no bulletproof ways, but at the moment it is super easy to do it because JSON is dumped for every page.

Is there at least way to hide the JSON?

**Comment by sergiodxa** [comment]
Only by removing the Scripts component in the root, but then you won’t get client side hydration. So no way without losing features.

With Single Data Fetch feature to be released behind a flag soon will change the data format to use turbo-stream and that way it will not be a simple JSON anymore.
